tests/static-analysis/assert-alpha.php

```diff
@@ -7,13 +7,41 @@
 /**
  * @psalm-pure
  *
- * @psalm-param non-empty-string $value
+ * @param mixed $value
  *
- * @psalm-return non-empty-string
+ * @return mixed
  */
-function consume(string $value): string
+function alpha($value)
 {
     Assert::alpha($value);
 
     return $value;
 }
+
+/**
+ * @psalm-pure
+ *
+ * @param mixed $value
+ *
+ * @return mixed
+ */
+function nullOrAlpha($value)
+{
+    Assert::nullOrAlpha($value);
+
+    return $value;
+}
+
+/**
+ * @psalm-pure
+ *
+ * @param mixed $value
+ *
+ * @return mixed
```

Comment:
`Assert::alpha` does not have an assertion, so it cannot assert anything. It's a limitation and because my suggestion was rejected (I don't agree with the reasoning though) - psalm won't ever support it.